content/0008-standalone-workflow-execution/rfc.md

```diff
@@ -0,0 +1,294 @@
+developer with a working cluster that can update Relay service deployments with
+new container images and run workflows without having to POST them to an
+instance of `relay-api`.
+
+In this version, we will add new commands to the Relay CLI. These commands will
+utilize packages from K3d, Kubernetes and other tools in the space that do a lot
+of the heavy lifting already.
+
+This version will require the developer to install a couple hard dependencies in
+order for `relay cluster` to function:
+
+* kubectl
+* docker
+
+## Drawbacks
+
+This might encourage developers to write less unit and integration tests. This
+must be avoided. We have to ensure all pull requests contain proper unit and
+integration tests for the service. It must be emphasized that `relay cluster` is
+not a replacement for continuous integration.
+
+## Rationale and alternatives
+
+### Why is this design the best in the space of possible designs?
+
+Running a local cluster means the cycle of writing code and running it against
+the relay system to see what happens can be quick. Developers can experiment a
+lot more with structure and risky changes. This also allows for verification of
+bug fixes to happen quicker.
+
+Other hopes for this system is catching subtle differences between two services
+that can be added to the test suite. For example: if you write code around the
+assumption that all services running are using the same message parsing module,
+you can quickly test this new code around various versions of the services. If
+you catch a failure between two versions, you can modify your code to support
+the different version and add test cases for that specifics. Or you can raise a
+red flag and start a proper deprecation of old message structures.
+
+### What other designs have been considered and what is the rationale for not choosing them?
+
+Other ideas that have been pitches are an always-on remote development cluster
+in GKE or the like. This has the drawback of being costly, has long development
+churn, and most-likely requires a CI build to run code.
+
+### What is the impact of not doing this?
+
+Developers will continue writing critical pieces of code and not running it
+locally before they commit. This causes a lot of frustration when subtle nuances
+between the services prevent the code from working properly. This can be
+effected by message payloads being slightly different between service versions,
+causing issues that weren't cause with testing code because of running version
+differences.
+
+### What specific risks are associated with this design?
+
+Over-engineering. We don't want to get distracted by creating yet another tool
+for general use. `relay` must always wrap existing tools and be opinionated around
+how to run Relay on a developer's machine.
+
+## Success criteria
+
+Developers can easily, without writing complex configuration, start a
+development cluster and run whatever version of a relay service they want.
```

Comment:
is there a second criteria here where workflow authors can run a workflow locally without connecting to the service?